### ISSUE TITLE
New chroot option and symlinks

### ISSUE THREAD

**Issue by jamieburchell** [dompdf/dompdf#2257]
I have the directory structure:

```
/www/project/current/httpdocs
/www/project/current/httpdocs/css
/www/project/current/httpdocs/assets > /www/project/shared/httpdocs/assets (symlink)
```

Some of the assets required for dompdf are in the CSS directory, others in assets. So, I set:

```
$options = new DompdfOptions;
$options->set('isRemoteEnabled', true);
$options->set('chroot', '/www/project/current/httpdocs');
```

Unfortunately, this meant images from the storage directory were failing to render in the PDF. Is this the intended behaviour? I can see why it might be for security reasons, but it doesn't help me here.

I ended up setting the chroot to `/www/project` instead, seeing as I don't think it's yet possible to provide multiple locations.

I already have `open_basedir` configured at PHP level; I'm not sure why this needs to be set at library level too?

TIA
Jamie

**Comment by bsweeney** [comment]
`open_basedir` won't help in some circumstances, such as certain multi-tenant installations. It's also important to remember that some users run in environments where they don't have full control of their PHP configuration. Layers of security provide better options to cover a wider range of installations. If a user is satisfied that the security provided at a different layer is sufficient then setting the chroot to be more open is a valid compromise.

FYI, #2232 will address the multi-root question.

**Comment by bsweeney** [comment]
Your point about breaking functionality has been made elsewhere. I freely admit that the way this change was released has been problematic. I welcome the input and we'll certainly consider how to handle this better in the future. Releasing changes like this as optionally enabled for a release or two is a good idea ... wish I had considered that before release.

**Comment by machek** [comment]
I am having exactly same issue. Discovered hard way, I will post, if I find solution for the issue, I hope will not end up with chroot = '/'

**Comment by bsweeney** [comment]
@machek that will be addressed in the next release, see #2232.
